<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>522</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>单轴步进系统控制工具</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QGridLayout" name="gridLayout_11" rowstretch="4,0,0,0">
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>模式选择</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_10">
      <property name="horizontalSpacing">
       <number>30</number>
      </property>
      <item row="0" column="3">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="checkBoxTwo">
        <property name="text">
         <string>寸动</string>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="checkBoxOne">
        <property name="text">
         <string>电机控制</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QCheckBox" name="checkBoxPoint">
        <property name="text">
         <string>点动</string>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="pushButtonSendMode">
        <property name="styleSheet">
         <string notr="true">QPushButton
{
	
	background-color: rgb(255, 85, 0);
}
QPushButton::pressed
{
	
	
	background-color: rgb(42, 218, 165);
}</string>
        </property>
        <property name="text">
         <string>发送</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout_4" columnstretch="1,0,0">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item row="0" column="1">
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>发送信息区:</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0" colspan="4">
         <widget class="QTextEdit" name="textEditSend"/>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBoxSendHex">
          <property name="text">
           <string>HEX</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBoxSendText">
          <property name="text">
           <string>文本显示</string>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="pushButtonSendSend">
          <property name="text">
           <string>发送</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="pushButtonSendClear">
          <property name="text">
           <string>清除发送</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QGroupBox" name="groupBoxInch_2">
       <property name="title">
        <string>点动</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="3" column="1">
         <widget class="QLineEdit" name="lineEditPointSpeed">
          <property name="text">
           <string/>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButtonPointAdd">
            <property name="text">
             <string>正向</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="pushButtonPointSub">
            <property name="text">
             <string>反向</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>速度：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="2">
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>x(10^-4)rad/s</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>接收信息区</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="2">
         <widget class="QPushButton" name="pushButtonClearRecv">
          <property name="text">
           <string>清空接收</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QTextEdit" name="textEditRecv">
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBoxRecvHex">
          <property name="text">
           <string>HEX</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBoxRecvText">
          <property name="text">
           <string>文本显示</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QGridLayout" name="gridLayout_9" columnstretch="1,1,1">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item row="0" column="0">
      <widget class="QGroupBox" name="leftTopGroupBox">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="title">
        <string>串口</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>波特率：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="comboBoxNum">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>64</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>128</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QPushButton" name="pushButtonUpdateCom">
          <property name="styleSheet">
           <string notr="true">background-color:rgb(127,255,212);</string>
          </property>
          <property name="text">
           <string>刷新串口</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="comboBoxBaund">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>115200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1200</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBoxName"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>串口号：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>细分数量：</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="pushButtonOpen">
          <property name="styleSheet">
           <string notr="true">background-color:rgb(127,255,212);</string>
          </property>
          <property name="text">
           <string>打开串口</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QPushButton" name="pushButtonReset">
          <property name="text">
           <string>复位</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QPushButton" name="pushButtonStopFast">
          <property name="text">
           <string>急停</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QGroupBox" name="groupBoxRight">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="title">
        <string>电机控制</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="1" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>角度：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBoxDirOne">
          <item>
           <property name="text">
            <string>正转</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>反转</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>方向：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="lineEditSpeedOne"/>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditAngleOne"/>
        </item>
        <item row="5" column="1">
         <widget class="QPushButton" name="pushButtonSetOne">
          <property name="text">
           <string>设置</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLineEdit" name="lineEditPulesOne">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(240, 240, 240);</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>脉冲数：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>转速：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>x(10^-4)rad</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>x(10^-4)rad/s</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>个</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QGroupBox" name="groupBoxRight_2">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="title">
        <string>寸动 (梯形加减速)</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>速度：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="comboBoxDirTwo">
          <item>
           <property name="text">
            <string>反转</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>正转</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEditLengthTwo"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>长度：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditSpeedTwo"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>方向：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>x(10^-2)mm</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>x(10^-4)rad/s</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_17">
     <property name="text">
      <string>Author: DuskyBook    Github : https://github.com/jinsechaochen/Epigynous-Computer-for-Single-axis-Stepper-Motor-System     Version: 2.0</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
